keps/sig-node/4815-dra-partitionable-devices/README.md

```diff
@@ -0,0 +1,3411 @@
+        copy-engines:
+          quantity: "7"
+        decoders:
+          quantity: "5"
+        encoders:
+          quantity: "0"
+        jpeg-engines:
+          quantity: "1"
+        ofa-engines:
+          quantity: "1"
+
+  - name: memory-slices-0
+    basic:
+      capacity:
+        memorySlice0:
+          quantity: "1"
+
+  - name: memory-slices-1
+    basic:
+      capacity:
+        memorySlice1:
+          quantity: "1"
+
+  - name: memory-slices-2
+    basic:
+      capacity:
+        memorySlice2:
+          quantity: "1"
+
+  - name: memory-slices-3
+    basic:
+      capacity:
+        memorySlice3:
+          quantity: "1"
+
+  - name: memory-slices-4
+    basic:
+      capacity:
+        memorySlice4:
+          quantity: "1"
+
+  - name: memory-slices-5
+    basic:
+      capacity:
+        memorySlice5:
+          quantity: "1"
+
+  - name: memory-slices-6
+    basic:
+      capacity:
+        memorySlice6:
+          quantity: "1"
+
+  - name: memory-slices-7
+    basic:
+      capacity:
+        memorySlice7:
+          quantity: "1"
+
+  - name: memory-slices-0-1
+    basic:
+      capacity:
+        memorySlice0:
+          quantity: "1"
+        memorySlice1:
+          quantity: "1"
+
+  - name: memory-slices-2-3
+    basic:
+      capacity:
+        memorySlice2:
+          quantity: "1"
+        memorySlice3:
+          quantity: "1"
+
+  - name: memory-slices-4-5
+    basic:
+      capacity:
+        memorySlice4:
+          quantity: "1"
+        memorySlice5:
+          quantity: "1"
+
+  - name: memory-slices-6-7
+    basic:
+      capacity:
+        memorySlice6:
+          quantity: "1"
+        memorySlice7:
+          quantity: "1"
+
+  - name: memory-slices-0-3
+    basic:
+      capacity:
+        memorySlice0:
+          quantity: "1"
+        memorySlice1:
+          quantity: "1"
+        memorySlice2:
+          quantity: "1"
+        memorySlice3:
+          quantity: "1"
+
+  - name: memory-slices-4-7
+    basic:
+      capacity:
+        memorySlice4:
+          quantity: "1"
+        memorySlice5:
+          quantity: "1"
+        memorySlice6:
+          quantity: "1"
+        memorySlice7:
+          quantity: "1"
+
+  - name: memory-slices-0-7
+    basic:
+      capacity:
+        memorySlice0:
+          quantity: "1"
+        memorySlice1:
+          quantity: "1"
+        memorySlice2:
+          quantity: "1"
+        memorySlice3:
+          quantity: "1"
+        memorySlice4:
+          quantity: "1"
+        memorySlice5:
+          quantity: "1"
+        memorySlice6:
+          quantity: "1"
+        memorySlice7:
+          quantity: "1"
+
+  devices:
+  - name: gpu-0
+    basic:
+      includes:
+      - name: gpu
+      - name: gpu-0-attributes
+      - name: memory-slices-0-7
+
+  - name: gpu-0-mig-1g.5gb-0
+    basic:
+      includes:
+      - name: gpu-0-mig-attributes
+      - name: mig-1g.5gb
+      - name: memory-slices-0
+      consumesCapacityFrom:
+      - name: gpu-0
```

Comment:
You already said that you want to change it, so perhaps this will become clear soon, but let me call it out explicitly here: it's not obvious to me *what* this `gpu-0-mig-1g.5gb-0` device consumes from `gpu-0`. Presumably any capacity defined for the device (whether it's in the device itself or in a mixin) gets consumed?

Reply:
Yeah, as it's defined right now, *all* capacity must be consumed from somewhere if this field is populated. There is no way to list capacities particular to this device that aren't consumed from elsewhere.

That said, there is (technically) nothing preventing you from listing yourself as a source here. That way you would pull all resources available on GPU 0 from there first, falling back to yourself for those that are only specific to you.

Reply:
> That said, there is (technically) nothing preventing you from listing yourself as a source here. 

Hmm, isn't consumption said to be recursive? Referencing yourself would lead to infinite recursion.

Reply:
Which might be an important point: allow a driver to publish something that can lead to infinite recursion should be prevented by validation.

Reply:
This is related to @thockin's comment here:
https://github.com/kubernetes/enhancements/pull/4874/files#r1795977481